beets/plugins.py

```diff
@@ -270,69 +265,99 @@ def helper(func: TFunc[Item]) -> TFunc[Item]:
-            except ImportError as exc:
-                # Again, this is hacky:
-                if exc.args[0].endswith(" " + name):
-                    log.warning("** plugin {0} not found", name)
-                else:
-                    raise
+            namespace = __import__(modname, None, None)
+        except ImportError as exc:
+            # Again, this is hacky:
+            if exc.args[0].endswith(" " + name):
+                log.warning("** plugin {0} not found", name)
             else:
-                for obj in getattr(namespace, name).__dict__.values():
-                    if (
-                        inspect.isclass(obj)
-                        and not isinstance(
-                            obj, GenericAlias
-                        )  # seems to be needed for python <= 3.9 only
-                        and issubclass(obj, BeetsPlugin)
-                        and obj != BeetsPlugin
-                        and not inspect.isabstract(obj)
-                        and obj not in _classes
-                    ):
-                        _classes.add(obj)
-
-        except Exception:
-            log.warning(
-                "** error loading plugin {}:\n{}",
-                name,
-                traceback.format_exc(),
-            )
+                raise
+        else:
+            for obj in getattr(namespace, name).__dict__.values():
+                if (
+                    inspect.isclass(obj)
+                    and not isinstance(
+                        obj, GenericAlias
+                    )  # seems to be needed for python <= 3.9 only
+                    and issubclass(obj, BeetsPlugin)
+                    and obj != BeetsPlugin
+                    and not inspect.isabstract(obj)
+                ):
+                    return obj()
+
+    except Exception:
+        log.warning(
+            "** error loading plugin {}:\n{}", name, traceback.format_exc()
+        )
+
+    return None
 
 
-_instances: dict[type[BeetsPlugin], BeetsPlugin] = {}
+_instances: list[BeetsPlugin] = []
 
 
-def find_plugins() -> list[BeetsPlugin]:
-    """Returns a list of BeetsPlugin subclass instances from all
-    currently loaded beets plugins. Loads the default plugin set
-    first.
+def load_plugins() -> None:
+    """Initialize the plugin system by loading all configured plugins.
+
+    Performs one-time plugin discovery and instantiation, storing loaded plugin
+    instances globally. Emits a pluginload event after successful initialization
+    to notify other components.
     """
-    if _instances:
-        # After the first call, use cached instances for performance reasons.
-        # See https://github.com/beetbox/beets/pull/3810
-        return list(_instances.values())
+    if not _instances:
+        names = get_plugin_names()
+        log.info("Loading plugins: {}", ", ".join(sorted(names)))
+        _instances.extend(filter(None, map(_get_plugin, names)))
```

Comment:
Yet another short question before I approve this:

We now omit the `and obj not in _classes` check. If a user now wants to load a plugin two times, what will happen? Is that an issue, will the plugin be loaded twice? 
I think there had to be a reason for this check.

Reply:
Awesome! Guess that works. Maybe warning the user would also be appropriate?